server/controllers/finance/reports/reportAccountsMultiple/index.js

```diff
@@ -0,0 +1,97 @@
+const _ = require('lodash');
+const q = require('q');
+const ReportManager = require('../../../../lib/ReportManager');
+
+const AccountsExtra = require('../../accounts/extra');
+const Exchange = require('../../exchange');
+const Currency = require('../../currencies');
+const Fiscal = require('../../fiscal');
+
+const TEMPLATE = './server/controllers/finance/reports/reportAccountsMultiple/report.handlebars';
+
+const { getAccountTransactions } = require('../reportAccounts/index');
+/**
+ * @method document
+ *
+ * @description
+ * Renders the PDF template for the Accounts Statement Report.
+ *
+ * The report contains the following information:
+ *  1. headers with the opening balance line and All general ledger transactions
+ *  for each account.  This opening balance line is
+ *  converted on the date of the `dateFrom` range.
+ *
+ */
+function document(req, res, next) {
+  let report;
+  const bundle = {};
+
+  const params = req.query;
+  params.user = req.session.user;
+  params.enterprise_id = req.session.enterprise.id;
+  params.isEnterpriseCurrency = (req.session.enterprise.currency_id === Number(params.currency_id));
+
+  try {
+    report = new ReportManager(TEMPLATE, req.session, params);
+  } catch (e) {
+    return next(e);
+  }
+
+  params.dateFrom = (params.dateFrom) ? new Date(params.dateFrom) : new Date();
+
+  // first, we look up the currency to have all the parameters we need
+  return Currency.lookupCurrencyById(params.currency_id)
+    .then(currency => {
+      _.extend(bundle, { currency });
+
+      // get the exchange rate for the opening balance
+      return Exchange.getExchangeRate(params.enterprise_id, params.currency_id, params.dateFrom);
+    })
+    .then(rate => {
+      bundle.rate = rate.rate || 1;
+      bundle.invertedRate = Exchange.formatExchangeRateForDisplay(bundle.rate);
+      return AccountsExtra.getOpeningBalanceForDate(params.account_id, params.dateFrom, false);
+    })
+    .then(balance => {
+      const { rate, invertedRate } = bundle;
+
+      const header = {
+        date            : params.dateFrom,
+        balance         : Number(balance.balance),
+        credit          : Number(balance.credit),
+        debit           : Number(balance.debit),
+        exchangedCredit : Number(balance.credit) * rate,
+        exchangedDebit : Number(balance.debit) * rate,
+        exchangedBalance : Number(balance.balance) * rate,
+        isCreditBalance : Number(balance.balance) < 0,
+        rate,
+        invertedRate,
+      };
+
+      _.extend(bundle, { header });
+
+      const { selectedAccounts } = params;
+      const accounts = [].concat(selectedAccounts);
+      return q.all(accounts.map(account => {
+        const parseAccount = JSON.parse(account);
```

Comment:
This statement tells me that you are not sending account IDs to the server, but JSON objects.  You should be sending an array of one or more account IDs to the server.  Like this:
```
GET some/report?account_ids=1,2,3
```
That way, there won't be any need to call `JSON.parse` on each value.